Src/ILGPU.Tests/GridOperations.cs

```diff
@@ -51,5 +52,29 @@ public void GridDimension(int xMask, int yMask, int zMask)
                 Verify(buffer, expected);
             }
         }
+
+        internal static void GridLaunchDimensionKernel(ArrayView<int> data)
+        {
+            data[0] = Grid.DimX;
+        }
+
+        // This test is one-dimensional and uses small sizes for the sake of passing
+        // tests on the CI machine, but on a machine with more threads it works
+        // for higher dimensions and higher sizes.
+        [Fact]
+        public void GridLaunchDimension()
+        {
+            using var buffer = Accelerator.Allocate<int>(1);
+            var kernel = Accelerator.LoadStreamKernel<ArrayView<int>>
+                (GridLaunchDimensionKernel);
+
+            kernel((1, 2), buffer.View); 
```

Comment:
ISSUE: Actually, I know I said it was fine, but on second thought, this is going to pass on OpenCL even with the old/broken code.

The original issue was that launching (10, 256) would cause `Grid.DimX` to return 2560 instead of 256.
In this case, launching (1, 2) would incorrectly return 2, regardless of whether the code was fixed or not.

Reply:
Why is this the case? Am I missing something here?

Reply:
Isn't the first number the Grid dimension? When we discussed, I believe we said the problem was it was returning 2560 instead of 10.

Reply:
Oops, you're right. The old code would return 2. The fixed code will return 1.
Don't mind me...